pallets/nfts/src/features/approvals.rs

```diff
@@ -157,6 +164,15 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			Item::<T, I>::get(collection, item).ok_or(Error::<T, I>::UnknownCollection)?;
 
 		if let Some(check_origin) = maybe_check_origin {
+			// Cannot revoke approvals for individual items when there are existing approvals to
+			// transfer all items in the collection owned by the origin.
+			ensure!(
+				CollectionApprovals::<T, I>::iter_prefix((collection, &check_origin))
+					.take(1)
+					.next()
+					.is_none(),
+				Error::<T, I>::DelegateApprovalConflict
```

Comment:
Tin please ensure you check your tests that you've written. The fact that after all these reviews these are still missing is a bit dissapointing.